frontend/src/app/main/refs.cljs

```diff
@@ -82,17 +82,32 @@
                (dm/get-in state [:dashboard-local :selected-project]))
              st/state))
 
+(defn- dashboard-extract-selected
+  [files selected]
+  (let [get-file #(get files %)
+        sim-file #(select-keys % [:id :name :project-id :is-shared])
+        xform    (comp (map get-file)
+                       (map sim-file))]
+    (->> (into #{} xform selected)
+         (d/index-by :id))))
+
+(def dashboard-selected-search
+  (l/derived (fn [state]
+               ;; we need to this because :dashboard-search-result is a list
+               ;; of maps and we need a map of maps (using :id as key).
+               (let [files (reduce (fn [m v] (assoc m (:id v) v)) {} (:dashboard-search-result state))]
```

Comment:
For thus purpose I guess you have the `d/index-by` helper

Reply:
I don't understand this, just below the last line (in line 99) I'm using `dashboard-extract-selected`, a function that uses `d/index-by` to extract the search files.

Reply:
This reduce looks like doing the same as d/index-by, or I'm wrong?

Reply:
Mmm... I don't think so, this reduce is converting the :dashboard-search-result, a list of files, into a map of files using it's :id as key.